torch/csrc/autograd/python_function.cpp

```diff
@@ -252,6 +327,21 @@ void PyNode::compiled_args(CompiledNodeArgs& args) {
   args.collect(f->materialize_non_diff_grads);
   args.collect(f->output_info);
   args.collect(f->input_info);
+
+  static PyObject* forward_cls_name =
+      PyUnicode_InternFromString("_forward_cls");
+  PyObject* forward_cls(PyObject_GetAttr(obj, forward_cls_name));
+  static PyObject* backward_name = PyUnicode_InternFromString("backward");
+  PyObject* backward(PyObject_GetAttr(forward_cls, backward_name));
+  int backward_idx =
+      args.add_backward(c10::SafePyObject(backward, getPyInterpreter()));
+
+  PyObject* saved_tensors(unpack_saved_variables(
+      f, [](const Variable& var) { return THPVariable_Wrap(var); }));
+  int saved_tensors_idx = args.add_saved_tensors(
+      c10::SafePyObject(saved_tensors, getPyInterpreter()));
+  TORCH_INTERNAL_ASSERT(backward_idx == saved_tensors_idx);
+  _backward_idx = backward_idx;
```

Comment:
Unpacking here has side effects.  We need to use the visitor pattern to visit the saved variables so that they get lifted into args to the graph.